src/openvic-simulation/interface/GFXObject.cpp

```diff
@@ -190,3 +136,178 @@ bool Actor::_fill_key_map(NodeTools::case_insensitive_key_map_t& key_map) {
+	);
+
+	return ret;
+}
+
+/* MapItems.gfx */
+Projection::Projection() : 
+	texture_file {}, size { 1 }, spin { 1 },
+	pulsating { false }, pulse_lowest { 1 }, pulse_speed { 1 },
+	additative { false }, expanding { 1 }, duration {}, fadeout {} {}
+
+//TODO: Verify...
+// whether pulseSpeed is fixedpoint_t or int
+// pulseSpeed doesn't seem to do anything, so assume fixed_point_t since its a speed
+//fadeout could be int, expect_int_bool, or fixed_point_t
+//fadeout seems not to do anything
+bool Projection::_fill_key_map(NodeTools::case_insensitive_key_map_t& key_map) {
+	bool ret = Object::_fill_key_map(key_map);
+
+	ret &= add_key_map_entries(key_map,
+		"textureFile", ONE_EXACTLY, expect_string(assign_variable_callback_string(texture_file)),
+		"size", ONE_EXACTLY, expect_fixed_point(assign_variable_callback(size)),
+		"spin", ONE_EXACTLY, expect_fixed_point(assign_variable_callback(spin)),
+		"pulsating", ONE_EXACTLY, expect_bool(assign_variable_callback(pulsating)),
+		"pulseLowest", ONE_EXACTLY, expect_fixed_point(assign_variable_callback(pulse_lowest)),
+		"pulseSpeed", ONE_EXACTLY, expect_fixed_point(assign_variable_callback(pulse_speed)),
+		"additative", ONE_EXACTLY, expect_bool(assign_variable_callback(additative)),
+		"expanding", ONE_EXACTLY, expect_fixed_point(assign_variable_callback(expanding)),
+		"duration", ZERO_OR_ONE, expect_fixed_point(assign_variable_callback(duration)),
+		"fadeout", ZERO_OR_ONE, expect_fixed_point(assign_variable_callback(fadeout))
+	);
+
+	return ret;
+}
+
+
+Billboard::Billboard() : 
+	texture_file {}, scale { 1 }, no_of_frames { 1 }, font_size { 7 },
+	offset {}, font {} {}
+
+bool Billboard::_fill_key_map(NodeTools::case_insensitive_key_map_t& key_map) {
+	bool ret = Object::_fill_key_map(key_map);
+
+	ret &= add_key_map_entries(key_map,
+		"texturefile", ONE_EXACTLY, expect_string(assign_variable_callback_string(texture_file)),
+		"noOfFrames", ZERO_OR_ONE, expect_int64(assign_variable_callback(no_of_frames)),
+		"scale", ONE_EXACTLY, expect_fixed_point(assign_variable_callback(scale)),
+		"font_size", ZERO_OR_ONE, expect_int64(assign_variable_callback(font_size)),
+		"offset2", ZERO_OR_ONE, expect_fvec3(assign_variable_callback(offset)),
+		"font", ZERO_OR_ONE, expect_string(assign_variable_callback_string(font))
+	);
+
+	return ret;
+}
+
+UnitStatsBillboard::UnitStatsBillboard() : 
+	texture_file {}, mask_file {}, effect_file {}, scale { 1 },
+	no_of_frames { 1 }, font_size { 7 }, font {} {}
+
+//TODO: Verify font_size is int
+bool UnitStatsBillboard::_fill_key_map(NodeTools::case_insensitive_key_map_t& key_map) {
+	bool ret = Object::_fill_key_map(key_map);
+
+	ret &= add_key_map_entries(key_map,
+		"textureFile", ONE_EXACTLY, expect_string(assign_variable_callback_string(texture_file)),
+		"mask", ONE_EXACTLY, expect_string(assign_variable_callback_string(mask_file)),
+		"effectFile", ONE_EXACTLY, expect_string(assign_variable_callback_string(effect_file)),		
```

Comment:
```suggestion
		"effectFile", ONE_EXACTLY, expect_string(assign_variable_callback_string(effect_file)),
```
Two stray tab characters at the end of this line